tests/test_cluster_s2_24_fix.py

```
@@ -0,0 +1,229 @@
+    async def test_worker_lease_cap_counts_active_only(self):
+        """Lease cap counts only active (non-expired) leases."""
+        mgr = ClusterManager()
+        worker = _make_worker("cap-worker", resources=[f"gpu-cuda-{i}" for i in range(12)])
+        await mgr.register_worker(worker)
+
+        for i in range(10):
+            lease = await mgr.claim_lease(f"cap-worker:gpu-cuda-{i}", caller=f"caller-{i}", ttl_seconds=0.001)
+            assert lease is not None
+
+        time.sleep(0.1)
+        active = [l for l in mgr._leases.values() if l.expires_at > time.time()]
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)